dev/benchmarking/ddl/1_05_table_public_language_dependencies.sql

```diff
@@ -0,0 +1,42 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+*/
+
+CREATE EXTENSION IF NOT EXISTS "citext";  -- type for case-insensitive text
+
+-- LANGUAGE_DEPENDENCIES
+CREATE TABLE IF NOT EXISTS public.language_dependencies
+(
+  language_dependencies_id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , dependencies jsonb NOT NULL DEFAULT '{}'::jsonb
+  , PRIMARY KEY (language_dependencies_id, benchmark_language_id)
+  , FOREIGN KEY (benchmark_language_id) REFERENCES public.benchmark_language
+);
+COMMENT ON TABLE public.language_dependencies
+  IS E'@name language_dependencies.\n'
+      'A JSON string mapping dependencies to their versions for each language.';
+COMMENT ON COLUMN public.language_dependencies.dependencies
+  IS 'For example: ''{"boost": "1.69", "conda": "", "numpy": "1.15"}''.';
```

Comment:
Yes, but other suggestions are desired!  The "benchmark_language" is intended to be uniquely tied to the "benchmark" definition (and the "environment" definition) and indicates the language *that the benchmark is written in.*

All other dependencies, be they C++ compilers, boost libraries, or whatever else all goes into the JSON object (key-value) in the column "public.language_dependencies.dependencies".

Please if this is a wrong interpretation of the intent let me know and I will happily fix it.
Or if you can think of some better way to describe the environment that's not my approach right now; I think it is kludgy but also can't think of anything better.

Reply:
Ok. Then I just don't think it's not useful to have a `benchmark_language_id` column here. The benchmark language is an attribute of the benchmark, not of the set of dependencies. It shouldn't be a problem if a C++ benchmark and a Python benchmark share the same dependencies, should it? (in all likelihood they won't, anyway ;-))

Reply:
> @tanyaschlusser We were wondering how it would be possible to hook this database into the CodeSpeed Web UI. Do you think it will be doable to create read-only views that mimick CodeSpeed's expected schema? For reference, the CodeSpeed schema is there:
> https://github.com/tobami/codespeed/blob/master/codespeed/models.py
> 
> No need to do it right now, just asking for feasibility.

Yes! I did this a few weeks ago at @aregm 's request for [ARROW-4354](https://issues.apache.org/jira/browse/ARROW-4354) 

It requires me to make a schema, named "codespeed" or "django", and populate it with views from the public schema that match Django's table naming convention. The only difference from a standard Django installation is that you would never run the `makemigrations` command because the database is already made. Also, depending on everyone's inclination, I could disable the admin interface and have all administration done directly on-database. (This would be my preference, since otherwise they would be manipulating views which would require me to write functions to handle.)

Reply:
Great, thank you. It is deleted.